From b9c084325a6b38c301ee498c9ef69f8321cfde87 Mon Sep 17 00:00:00 2001
From: Mingxing Cao <caomingxing@huawei.com>
Date: Wed, 19 Apr 2023 11:52:56 +0800
Subject: [PATCH 0882/1646] [Huawei] pci: fixup yh bar size

Offering: HULK
optical inclusion
category: feature
bugzilla: NA

-------------------------------------------------

Use DECLARE_PCI_FIXUP_HEADER to correct yh bar size.

Signed-off-by: Mingxing Cao<caomingxing@huawei.com>
---
 drivers/pci/quirks.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index e30c2b16c67e..bdfa82c4f286 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -5905,3 +5905,15 @@ DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_HUAWEI, PCIE_DEVICE_ID_HISI_5991,
 			 pci_quirk_hisi_fixup_bar);
 DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_HUAWEI, PCIE_DEVICE_ID_HISI_5887,
 			 pci_quirk_hisi_fixup_bar);
+
+static void pci_quirk_188b_fixup_bar(struct pci_dev *dev)
+{
+#define PCIE_MEM_SPACE_SIZE 0x4000000 // 64MB
+
+	dev->resource[0].end = dev->resource[0].start + PCIE_MEM_SPACE_SIZE - 1;
+
+	pci_info(dev, "force resize pcie bar0\n");
+}
+
+DECLARE_PCI_FIXUP_HEADER(0x188b, 0x0008, pci_quirk_188b_fixup_bar);
+
-- 
2.17.1

